src/modal.service.ts

```diff
@@ -0,0 +1,30 @@
+import {Injectable} from '@angular/core';
+
+import {ModalComponent} from './modal.component';
+
+@Injectable()
+export class ModalService {
+  private modals: Array<ModalComponent> = [];
+
+  constructor() {
+    window.addEventListener('keydown', (event) => {
```

Comment:
window.addEventListener requires a requires a [removeEventListener](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/removeEventListener) to prevent memory leaks.

Reply:
thanks for pointing this. Anyway, I will delete the service.

What about remove service related commits. And let closing only the last modal as a TODO. At least for me, in this moment, is not a must and it will require some discussion.